rastervision_pytorch_backend/rastervision/pytorch_backend/examples/object_detection/cowc_potsdam.py

```diff
@@ -9,21 +9,63 @@
 from rastervision.core.analyzer import *
 from rastervision.pytorch_backend import *
 from rastervision.pytorch_learner import *
-from rastervision.pytorch_backend.examples.utils import get_scene_info, save_image_crop
+from rastervision.pytorch_backend.examples.utils import (get_scene_info,
+                                                         save_image_crop)
 
+TRAIN_IDS = [
+    '2_10', '2_11', '2_12', '2_14', '3_11', '3_13', '4_10', '5_10', '6_7',
+    '6_9'
+]
+VAL_IDS = ['2_13', '6_8', '3_10']
 
-def get_config(runner, raw_uri, processed_uri, root_uri, test=False):
-    train_ids = [
-        '2_10', '2_11', '2_12', '2_14', '3_11', '3_13', '4_10', '5_10', '6_7',
-        '6_9'
-    ]
-    val_ids = ['2_13', '6_8', '3_10']
+
+def get_config(runner,
+               raw_uri: str,
+               processed_uri: str,
+               root_uri: str,
+               nochip: bool,
+               test: bool = False) -> ObjectDetectionConfig:
+    """Generate the pipeline config for this task. This function will be called
+    by RV, with arguments from the command line, when this example is run.
+
+    Args:
+        runner (Runner): Runner for the pipeline. Will be provided by RV.
+        raw_uri (str): Directory where the raw data resides
+        processed_uri (str): Directory for storing processed data. 
+                             E.g. crops for testing.
+        root_uri (str): Directory where all the output will be written.
+        nochip (bool, optional): If True, read directly from the TIFF during
+            training instead of from pre-generated chips. The analyze and chip
+            commands should not be run, if this is set to True. Defaults to
+            False.
+        test (bool, optional): If True, does the following simplifications:
+            (1) Uses only the first 2 scenes
+            (2) Uses only a 2000x2000 crop of the scenes
+            (3) Enables test mode in the learner, which makes it use the
+                test_batch_sz and test_num_epochs, and also halves the img_sz.
+            Defaults to False.
+
+    Returns:
+        ObjectDetectionConfig: A pipeline config.
+    """
+    train_ids = TRAIN_IDS
+    val_ids = VAL_IDS
 
     if test:
-        train_ids = train_ids[0:1]
-        val_ids = val_ids[0:1]
+        train_ids = train_ids[:2]
+        val_ids = val_ids[:2]
 
-    def make_scene(id):
+    chip_sz = 600
+    img_sz = 256
+    if nochip:
+        raster_source_persist = True
```

Comment:
It seems odd that the user needs to specify whether the raster source should be persisted. This seems like the sort of thing that should be determined automatically. I see how this could be difficult though because the raster source is constructed before the PyTorch Dataset. Maybe when the PyTorch Dataset gets the reference to the scenes, it could call a method on them which will set the persist flags.

Reply:
I agree. That makes sense.

Reply:
I have removed `persist` in favor of modifying `ActivateMixin`. The user no longer has to specify anything to change raster source behavior.

In a similar vein, `ChipClassificationLabelSourceConfig.lazy` is now set to `True` automatically. The point of the param being to disable inferring labels in the constructor (to avoid doing it twice).